### PR TITLE
Issue #4 | To set up a new shopping list

### LINKED ISSUE #4: 4. As a user, I want to set up a new shopping list so I can start tracking purchased items.
## Summary 

A shopping list is a set of items associated with the owner's unique id and the name of the shopping list. Users need to be able to create new shopping lists. To do this, the `Home` view should present them with a form that allows them to enter the name of their list. Additionally, we need to do a few things to create that list:

1. Save the token to `localStorage`
2. Save the list to our database using the `createList` function in `firebase.js` 
3. Show the user the list view
## Acceptance criteria
UI-related tasks:
- [x] The `Home` view displays a form that allows users to enter the name of a shopping list and then create a list with that name.
- [x] The input that accepts the name of the item has a semantic `label` element associated with it
- [x] The user can submit this form with both the mouse and the `Enter` key  
- [x]  When the user submits the form, they see a message indicating that the list either was or was not created and saved to the database.

Data-related tasks:
- [x] Clicking the button uses the `createList` function in `src/api/firebase.js` to create a new shopping list for the user.
- [x] The shopping list path is stored in local storage using the `setListPath` function.
- [x] Once the list has been created and saved, the user is redirected to the `List` view.

## Notes
- The `createList` function takes **three** arguments: the current user's id, the current user's email, and the name of the shopping list.
- The `setListPath` function takes **one** argument: the new list token.
- If you get stuck trying to redirect the user, check out the [useNavigate](https://reactrouter.com/en/6.14.2/hooks/use-navigate) hook from `react-rotuer`.

### PR DESCRIPTION
**Pull request by shuveksha-tuladhar**
## Description

- A labeled form was created on the Home page which sets the newList state. 
- The submitted form calls createList function using userId, userEmail, and adds a new shopping list to the database.
- createList function returns the path to the newList. 
- setListPath is used to update user entered name of their list and save the token to localStorage. 
- The newly added shopping list can be seen on the Home page.
- Finally, handleSubmit navigates to the List page of the newly created list. 

## Related Issue

- closes #4 

## Acceptance Criteria

UI-related tasks:

- [x] The Home view displays a form that allows users to enter the name of a shopping list and then create a list with that name.
- [x] The input that accepts the name of the item has a semantic label element associated with it
- [x] The user can submit this form with both the mouse and the Enter key
- [x] When the user submits the form, they see a message indicating that the list either was or was not created and saved to the database.

Data-related tasks:

- [x] Clicking the button uses the createList function in src/api/firebase.js to create a new shopping list for the user.
- [x] The shopping list path is stored in local storage using the setListPath function.
- [x] Once the list has been created and saved, the user is redirected to the List view.

## Type of Changes

- enhancement 
- create operation

### Before

image

### After

image

image

image

## Testing Steps / QA Criteria

1. Sign in with your email ID. 
2. Try clicking submit with an empty input. You should get an alert.
3. Type in a shopping list name and either click on _Submit_ button or press _Enter_.
4. You should see a success alert and navigate to _/list_ page.

### CODE REVIEW

File: src/views/Home.jsx [hunk 1]
```diff
@@ -1,15 +1,43 @@
 import './Home.css';
 import { SingleList } from '../components';
-export function Home({ data, setListPath }) {
+import { useState } from 'react';
+import { createList } from '../api';
+import { useNavigate } from 'react-router-dom';
+
+export function Home({ data, setListPath, user }) {
+	const navigate = useNavigate();
```
Comment (dterceroparker) [inline src/views/Home.jsx:8]: **👏 praise:** such a great idea to use optional chaining for error prevention!‏

**Review by dterceroparker — APPROVED** [review]
** 🦋 praise:** ‏Looks great and test worked. Great idea to use optional chaining for error prevention!

**Review by stacy-tech — APPROVED** [review]
Everything looks great! The code is well structured and the tests run successfully. Awesome job guys! Consider it approved! 🙌🏾

**Review by meganesu — APPROVED** [review]
Tested out the deploy preview, looks like it's all working as expected! Great work y'all 🥳

**Pull request merged**